addons/web/static/src/js/core/popover.js

```diff
@@ -0,0 +1,269 @@
+odoo.define('web.Popover', function() {
+    'use strict';
+
+    const { Component, hooks, misc, QWeb } = owl;
+    const { Portal } = misc;
+    const { useExternalListener, useRef, useState } = hooks;
+
+    /**
+     * Popover
+     *
+     * Represents a bootstrap-style popover handled with pure JS. The popover
+     * will be visually bound to its `target` using an arrow-like '::before'
+     * CSS pseudo-element.
+     * @extends Component
+     **/
+    class Popover extends Component {
+        /**
+         * @param {Object} props
+         * @param {String} props.position='bottom' 'top', 'bottom', 'left' or 'right'
+         * @param {String} [props.title]
+         */
+        constructor() {
+            super(...arguments);
+            this.popoverRef = useRef('popover');
+            this.orderedPositions = ['top', 'right', 'bottom', 'left'];
+            this.state = useState({
+                displayed: false,
+            })
+            // without event capture, scroll listener on children nodes will prevent
+            // this handler from being fired
+            useExternalListener(document, 'scroll', this.show, {
+                capture: true,
+            });
+            useExternalListener(window, 'resize', this.show);
+        }
+
+        patched() {
+            super.patched(...arguments);
+            this.show();
+        }
+
+        /**
+         * Show the popover according to its props. This method will try to position the
+         * popover as requested (according to the `position` props). If the requested position
+         * does not fit the viewport, other positions will be tried in a clockwise order starting
+         * a the requested position (e.g. starting from left: top, right, bottom). If no position
+         * is found that fits the viewport, 'bottom' is used.
+         */
+        show() {
+            if (!this.state.displayed) {
+                return;
+            }
+            // get the target from the dom
+            // we don't want to do this early, since the target might not
+            // be in the dom when the component is instanciated
+            if (!this.constructor._isInViewport(this.el)) {
+                // target is no longer in the viewport, close the popover
+                return this._close();
+            }
+
+            const positionIndex = this.orderedPositions.indexOf(
+                this.props.position
+            );
+
+            const positioningData = this.constructor._computePositioningData(
+                this.popoverRef.el,
+                this.el
+                );
+            // remove all positioning classes
+            this.orderedPositions.forEach((pos) => {
```

Comment:
no i'm gonna keep that after finding the correct position; has otherwise it triggers a visible repaint for some reason. I think the delay between removing the class and re-adding them should be done 'as quickly as possible' to avoid repaints.

Reply:
![f39ef527-09a5-4812-86cd-4719170a1021(1)](https://user-images.githubusercontent.com/9530767/79208708-b4886380-7e42-11ea-9046-7e32f0775993.gif)

Reply:
i don't know why it merged my previous capture with the one i did just now. Only the first 5sec are relevant :p

Reply:
weird... There should be something asynchronous somewhere

Reply:
(but ok, you can keep it as it was I guess)